### ISSUE TITLE
Including "down" server entries can lead to "no live upstreams while connecting to upstream"

### ISSUE THREAD

**Issue by wlonkly** [nginx-proxy/nginx-proxy#1495]
When the host on which nginx-proxy is running has multiple networks, nginx-proxy's template generates an upstream on which some networks have an upstream marked as `down`, like:

```
upstream music.home.mati.ca {
				## Can be connected with "bridge" network
			# lms
			server 172.17.0.2:9000;
				# Cannot connect to network of this container
				server 127.0.0.1 down;
}
```

This is fine when everything works, but if the connectable upstream (on `bridge` in my example) has _any_ problems (like, say, a single 502 error), it triggers nginx's `max_fails`/`fail_timeout` behavior, and that upstream is marked as down for several seconds, with the error `no live upstreams while connecting to upstream`.

However, nginx actually has a fallback case that when an upstream only has _one_ `server` listed, then it skips the "no live upstreams" case, on the assumption that with only one upstream it's better to just keep trying it. 

Unfortunately, in the generated template above, nginx thinks it has _two_ `server`s listed, even though one is `down`, so the fallback case doesn't apply. (nginx should probably not do this when an upstream is marked `down`, but here we are.)

There's two options I think we could do to fix this:

* Don't output the `down` entry -- but I think this will produce an invalid nginx config if a container has _no_ valid endpoints, and we want nginx to keep working in that case, since it's presumably serving other containers; we'd have to handle the case where there are no upstreams at all and only output a single `down` upstream in that case
* Hardcode `max_fails` to 0 on every `server`, the idea being that we're not expecting there to be load-balancing behind `nginx-proxy`, just local upstream containers.

(I lean towards the second thing because it's easier templating, but I also suspect people use nginx-proxy for things I've never considered.)

**Comment by wlonkly** [comment]
Quick update: I tried the `max_fails=0` option locally: before, I had about 20 502 errors per page load, after I had one, which was a legit failure in the upstream container.

**Comment by ChaseGuru** [comment]
For some reason I get this issue due to "down".

```
proxy_1        | nginx.1    | 2020/10/05 20:05:08 [error] 75#75: *1 no live upstreams while connecting to upstream, client: 24.11.40.134, server: mysite.com, request: "GET /admin/login/?next=/admin/ HTTP/2.0", upstream: "http://mysite.com/admin/login/?next=/admin/", host: "mysite.com"
proxy_1        | nginx.1    | mysite.com 24.11.40.134 - - [05/Oct/2020:20:05:08 +0000] "GET /admin/login/?next=/admin/ HTTP/2.0" 502 552 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36"
proxy_1        | nginx.1    | 2020/10/05 20:05:09 [error] 75#75: *1 no live upstreams while connecting to upstream, client: 24.11.40.134, server: mysite.com, request: "GET /admin/login/?next=/admin/ HTTP/2.0", upstream: "http://mysite.com/admin/login/?next=/admin/", host: "mysite.com"
proxy_1        | nginx.1    | mysite.com 24.11.40.134 - - [05/Oct/2020:20:05:09 +0000] "GET /admin/login/?next=/admin/ HTTP/2.0" 502 552 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36"
```
But once I remove the "down" inside my `/etc/nginx/conf.d/default.conf` upstream, and do `nginx -s reload` inside the containre it serves my site

results after change:
```
proxy_1        | nginx.1    | mysite.com 24.11.40.134 - - [05/Oct/2020:20:08:12 +0000] "GET /admin/login/?next=/admin/ HTTP/2.0" 200 2194 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36"
api_1          | [pid: 19|app: 0|req: 2/7] 172.20.0.3 () {54 vars in 1052 bytes} [Mon Oct  5 20:08:12 2020] GET /admin/login/?next=/admin/ => generated 2194 bytes in 83 msecs (HTTP/1.1 200) 9 headers in 461 bytes (1 switches on core 0)
```

**Comment by wlonkly** [comment]
@ChaseGuru Yep, that is this bug in a nutshell. The reason nginx-proxy adds `down` entries, from what I can tell, is to make sure that there is never an _empty_ `upstream` which is an nginx.conf syntax error. My PR above makes it so that the `down` entries don't trigger the "no live upstreams" issue.

**Comment by PipeItToDevNull** [comment]
I just started encountering this issue on a Nextcloud 19.0.3 container behind this proxy. I have been using a `proxy` and `backend` network for a long time without issue, now the `127.0.0.1 Down` entries are breaking everything.

**Comment by wlonkly** [comment]
@PipeItToDevNull Not sure about the maintenance situation on this repo, but to hold you over, try [the max-fails branch in my fork](https://github.com/wlonkly/nginx-proxy/tree/max_fails)? 

Also, did you (inadvertently?) add a network to docker recently? `docker network ls` will list them, and I _think_ the `down` behavior is only when there exists a network the container is not using.

**Comment by PipeItToDevNull** [comment]
Looking at this again tonight, it might not be related. I think I am breaking in other places.

**Comment by buchdag** [comment]
Fixed by #1667
